### PR TITLE
Ethernet: examples and readme update

### PR DESCRIPTION
**Pull request by pennam**
Add EthernetConnectionHandler in library examples and the list of ethernet enabled boards in the readme.

Bonus: 
https://github.com/arduino-libraries/ArduinoIoTCloud/commit/0802445d458a84cebdb9ac306beae4cd0395544a Make Basic example build for all ESP32 boards

https://github.com/arduino-libraries/ArduinoIoTCloud/commit/7c4eb0938a463446974629938c3efc52bdfde5b9 Align header format of library examples

### CODE REVIEW

File: examples/ArduinoIoTCloud-Advanced/arduino_secrets.h [hunk 1]
```diff
@@ -35,3 +35,11 @@
   #define SECRET_LOGIN ""
   #define SECRET_PASS ""
 #endif
+
+/* Portenta H7 + Ethernet shield */
+#if defined(BOARD_HAS_ETHERNET)
+  #define SECRET_OPTIONAL_IP ""
```
Comment (aentinger) [inline examples/ArduinoIoTCloud-Advanced/arduino_secrets.h:41]: Why are those secrets called `_OPTIONAL_`?

File: examples/ArduinoIoTCloud-Basic/ArduinoIoTCloud-Basic.ino [hunk 2]
```diff
@@ -6,18 +6,18 @@
...
-  This sketch is compatible with:
+  The full list of compatible boards can be found here:
    - https://github.com/arduino-libraries/ArduinoIoTCloud#what
 */
 
 #include "arduino_secrets.h"
 #include "thingProperties.h"
 
-#if defined(ESP32)
+#if !defined(LED_BUILTIN)
```
Comment (aentinger) [inline examples/ArduinoIoTCloud-Basic/ArduinoIoTCloud-Basic.ino:20]: Yes, that's much better :rofl: I'm afraid I was the author of the original implementation =)

File: examples/ArduinoIoTCloud-DeferredOTA/thingProperties.h [hunk 3]
```diff
@@ -34,4 +35,9 @@ void initProperties() {
   LoRaConnectionHandler ArduinoIoTPreferredConnection(SECRET_APP_EUI, SECRET_APP_KEY, _lora_band::EU868, NULL, _lora_class::CLASS_A);
 #elif defined(BOARD_HAS_NB)
   NBConnectionHandler ArduinoIoTPreferredConnection(SECRET_PIN, SECRET_APN, SECRET_LOGIN, SECRET_PASS);
+#elif defined(BOARD_HAS_ETHERNET)
+  /* DHCP mode */
+  //EthernetConnectionHandler ArduinoIoTPreferredConnection;
+  /* Manual mode. It will fallback in DHCP mode if SECRET_OPTIONAL_IP is invalid or equal to "0.0.0.0" */
```
Comment (aentinger) [inline examples/ArduinoIoTCloud-DeferredOTA/thingProperties.h:41]: Okay, the parameter is optional when using DHCP. But maybe a better naming would be `SECRET_MANUAL_IP`. Also its not really a secret, just a configuration item. Maybe `ETHERNET_MANUAL_CONFIG_IP`?

**Review by aentinger — APPROVED** [review]
LGTM :+1: Just some thought impulses re naming.

**Comment by pennam** [comment]
Thanks @aentinger . I've used the same naming used when exporting a sketch from the cloud editor.

This defines are indeed not secrets, but it wasn't possible to remove the `SECRET_` because cloud side are handled as if they were. The `_OPTIONAL_` part was added because the user may want or not to customize ethernet parameters. If it wont customize them the interface will work in DHCP mode.

Another :thinking:  fact is they are defined into the arduino_secrets.h file and the filename is not really appropriate in case of ethernet configuration parameter.

I feel like renaming only, would not be the best solution: maybe we should also move them in the thingProperties.h file. What do you think?

```
#if defined(BOARD_HAS_ETHERNET)
  /* DHCP mode */
  //EthernetConnectionHandler ArduinoIoTPreferredConnection;
  /* Manual mode. It will fallback in DHCP mode if SECRET_OPTIONAL_IP is invalid or equal to "0.0.0.0" */
  #define ETHERNET_MANUAL_CONFIG_IP ""
  #define ETHERNET_MANUAL_CONFIG_DNS ""
  #define ETHERNET_MANUAL_CONFIG_GATEWAY ""
  #define ETHERNET_MANUAL_CONFIG_NETMASK ""
  EthernetConnectionHandler ArduinoIoTPreferredConnection(ETHERNET_MANUAL_CONFIG_IP, ETHERNET_MANUAL_CONFIG_DNS, ETHERNET_MANUAL_CONFIG_GATEWAY, ETHERNET_MANUAL_CONFIG_NETMASK);
#elif defined(BOARD_HAS_WIFI)
  WiFiConnectionHandler ArduinoIoTPreferredConnection(SECRET_SSID, SECRET_PASS);
#elif defined(BOARD_HAS_GSM)
  GSMConnectionHandler ArduinoIoTPreferredConnection(SECRET_PIN, SECRET_APN, SECRET_LOGIN, SECRET_PASS);
#elif defined(BOARD_HAS_LORA)
  LoRaConnectionHandler ArduinoIoTPreferredConnection(SECRET_APP_EUI, SECRET_APP_KEY, _lora_band::EU868, NULL, _lora_class::CLASS_A);
#elif defined(BOARD_HAS_NB)
  NBConnectionHandler ArduinoIoTPreferredConnection(SECRET_PIN, SECRET_APN, SECRET_LOGIN, SECRET_PASS);
#endif
```

File: examples/ArduinoIoTCloud-DeferredOTA/thingProperties.h (hunk 3 above)
Comment (aentinger) [reply examples/ArduinoIoTCloud-DeferredOTA/thingProperties.h:41]: Thank you for your [clarification](https://github.com/arduino-libraries/ArduinoIoTCloud/pull/342#issuecomment-1306877015). I thought so. In this case there's not much you do I suppose :shrug:.

**Pull request merged**